server/parser.go

```diff
@@ -389,6 +389,17 @@ func (c *client) parse(buf []byte) error {
 				if err := c.processPub(arg); err != nil {
 					return err
 				}
+				// Check if we have and account mappings or tees or filters.
+				// FIXME(dlc) - Probably better way to do this.
+				// Could add in cache but will be tricky since results based on pub subject are dynamic
+				// due to wildcard matching and weight sets.
+				if c.kind == CLIENT && c.in.flags.isSet(hasMappings) {
+					old := c.pa.subject
+					c.selectMappedSubject()
+					if trace && !bytes.Equal(old, c.pa.subject) {
```

Comment:
other way would be for selectMappedSubject() to return a boolean to indicate that an actual mapping took place.

Reply:
I like that idea. Will incorporate.